foundry/mouseInteractionManager.d.ts

```diff
@@ -1,213 +1,257 @@
 /**
  * Handle mouse interaction events for a Canvas object.
- *
  * There are three phases of events: hover, click, and drag
+ * @typeParam O - Canvas object this instance handles events for
+ * @typeParam T - Target object for mouseinteraction events. Generally a `ControlIcon` or `O`.
+ *                (default: `O`)
  *
  * Hover Events:
- *
- *      _handleMouseOver
- *              action: hoverIn
- *      _handleMouseOut
- *              action: hoverOut
+ * _handleMouseOver
+ *  action: hoverIn
+ * _handleMouseOut
+ *  action: hoverOut
  *
  * Left Click and Double-Click
- *
- *      _handleMouseDown
- *              action: clickLeft
- *              action: clickLeft2
+ * _handleMouseDown
+ *  action: clickLeft
+ *  action: clickLeft2
  *
  * Right Click and Double-Click
- *
- *      _handleRightDown
- *              action: clickRight
- *              action: clickRight2
+ * _handleRightDown
+ *  action: clickRight
+ *  action: clickRight2
  *
  * Drag and Drop
- *
- *      _handleMouseMove
- *              action: dragLeftStart
- *              action: dragLeftMove
- *              action: dragRightStart
- *              action: dragLeftMove
- *      _handleMouseUp
- *              action: dragLeftDrop
- *              action: dragRightDrop
- *      _handleDragCancel
- *              action: dragLeftCancel
- *              action: dragRightCancel
+ * _handleMouseMove
+ *  action: dragLeftStart
+ *  action: dragLeftMove
+ *  action: dragRightStart
+ *  action: dragLeftMove
+ * _handleMouseUp
+ *  action: dragLeftDrop
+ *  action: dragRightDrop
+ * _handleDragCancel
+ *  action: dragLeftCancel
+ *  action: dragRightCancel
  */
-declare class MouseInteractionManager {
-  callbacks: object;
+declare class MouseInteractionManager<O extends PIXI.Container = PIXI.Container, T extends PIXI.Container = O> {
+  constructor(
```

Comment:
I think we should probably add the default values in JSDoc for the constructor.

Reply:
All of the default values here are `{}`, but I can add them if you think it's needed.